### PR TITLE
Adding HTTPAnalyticsResponseCodeFilter Widget

### PR DESCRIPTION
**Pull request by Niveathika92**
## Purpose
Create a widget that allows the user to select the filters in response code analytics page  to in HTTP Dashboard (HTTP Analytics Solution)

## Goals
Use the multi-select component to showcase data that was queried from the database using SiddhiAppProvider

## Approach
responsecodefilter

## User stories
As a user,
I want to be able to filter data based on service,
So that HTTP Server can be monitored.

## Release note
N/A

## Documentation
N/A

## Training
N/A

## Certification
N/A

## Marketing
N/A

## Automation tests
N/A

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples
N/A

## Related PRs
N/A

## Migrations (if applicable)
N/A

## Test environment
Chrome 64.0.3282.140
Firefox 60.0.1
 
## Learning
N/A

### CODE REVIEW

File: modules/solutions/http-analytics/widgets/HTTPAnalyticsResponseCodeFilter/src/HTTPAnalyticsResponseCodeFilter.jsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+        );
+    }
+}
+
+HTTPAnalyticsResponseCodeFilter.propTypes = {
+    classes: PropTypes.object.isRequired,
+};
+
+global.dashboard.registerWidget("HTTPAnalyticsResponseCodeFilter", HTTPAnalyticsResponseCodeFilter);
+
```
Comment (lasanthaS) [inline modules/solutions/http-analytics/widgets/HTTPAnalyticsResponseCodeFilter/src/HTTPAnalyticsResponseCodeFilter.jsx]: Remove unnecessary lines

File: modules/solutions/http-analytics/widgets/HTTPAnalyticsResponseCodeFilter/src/resources/dataProviderConf.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    "query": "from ServiceTable select *",
+    "publishingInterval": 60
+  }
+}
```
Comment (lasanthaS) [inline modules/solutions/http-analytics/widgets/HTTPAnalyticsResponseCodeFilter/src/resources/dataProviderConf.json]: Line break

Comment (Niveathika92) [reply modules/solutions/http-analytics/widgets/HTTPAnalyticsResponseCodeFilter/src/resources/dataProviderConf.json]: Fixed in 1b5b477

File: modules/solutions/http-analytics/widgets/HTTPAnalyticsResponseCodeFilter/src/HTTPAnalyticsResponseCodeFilter.jsx (hunk 1 above)
Comment (Niveathika92) [reply modules/solutions/http-analytics/widgets/HTTPAnalyticsResponseCodeFilter/src/HTTPAnalyticsResponseCodeFilter.jsx]: Fixed in 1b5b477

Comment (Niveathika92) [reply modules/solutions/http-analytics/widgets/HTTPAnalyticsResponseCodeFilter/src/HTTPAnalyticsResponseCodeFilter.jsx]: Fixed in b817e41

**Pull request merged**